warehouse/query-core/src/main/java/datawave/query/predicate/EventDataQueryFieldFilter.java

```diff
@@ -1,34 +1,47 @@
 package datawave.query.predicate;
 
 import com.google.common.collect.Sets;
-import datawave.query.attributes.Document;
 import datawave.query.jexl.JexlASTHelper;
 import org.apache.accumulo.core.data.Key;
-import org.apache.accumulo.core.data.PartialKey;
 import org.apache.accumulo.core.data.Range;
 import org.apache.commons.jexl2.parser.ASTIdentifier;
 import org.apache.commons.jexl2.parser.ASTJexlScript;
 
+import javax.annotation.Nullable;
 import java.util.Map;
 import java.util.Set;
+import java.util.SortedSet;
 
 /**
  * This filter will filter event data keys by only those fields that are required in the specified query.
  */
-public class EventDataQueryFieldFilter extends KeyProjection implements EventDataQueryFilter {
+public class EventDataQueryFieldFilter implements EventDataQueryFilter {
     private Set<String> nonEventFields;
 
-    public EventDataQueryFieldFilter() {
-        super();
-        // empty white list and black list
-    }
+    private final KeyProjection projection;
 
     public EventDataQueryFieldFilter(EventDataQueryFieldFilter other) {
-        super(other);
         this.nonEventFields = other.nonEventFields;
         if (other.document != null) {
             document = new Key(other.document);
         }
+        this.projection = other.getProjection();
+    }
+
+    /**
+     * Initiate with nothing
+     */
+    public EventDataQueryFieldFilter() {
+        this.projection = new KeyProjection();
+    }
+
+    /**
+     * Initiate from a KeyProjection
+     *
+     * @param includedFields
```

Comment:
I missed this in yesterday's review. Update the param name and provide a description to fix the doc build.